crates/turbopack-ecmascript/src/references/unreachable.rs

```diff
@@ -29,20 +45,199 @@ impl CodeGenerateable for Unreachable {
         &self,
         _context: Vc<Box<dyn ChunkingContext>>,
     ) -> Result<Vc<CodeGeneration>> {
-        let path = self.path.await?;
-        let visitors = [
-            // Unreachable might be used on Stmt or Expr
-            create_visitor!(exact path, visit_mut_expr(expr: &mut Expr) {
-                *expr = quote!("(\"TURBOPACK unreachable\", undefined)" as Expr);
-            }),
-            create_visitor!(exact path, visit_mut_stmt(stmt: &mut Stmt) {
-                // TODO(WEB-553) walk ast to find all `var` declarations and keep them
-                // since they hoist out of the scope
-                *stmt = quote!("{\"TURBOPACK unreachable\";}" as Stmt);
-            }),
-        ]
-        .into();
+        let range = self.range.await?;
+        let visitors = match &*range {
+            AstPathRange::Exact(path) => {
+                [
+                    // Unreachable might be used on Stmt or Expr
+                    create_visitor!(exact path, visit_mut_expr(expr: &mut Expr) {
+                        *expr = quote!("(\"TURBOPACK unreachable\", undefined)" as Expr);
+                    }),
+                    create_visitor!(exact path, visit_mut_stmt(stmt: &mut Stmt) {
+                        // TODO(WEB-553) walk ast to find all `var` declarations and keep them
+                        // since they hoist out of the scope
+                        let mut replacement = Vec::new();
+                        replacement.push(quote!("\"TURBOPACK unreachable\";" as Stmt));
+                        ExtractDeclarations {
+                            stmts: &mut replacement,
+                        }.visit_mut_stmt(stmt);
+                        *stmt = Stmt::Block(BlockStmt {
+                            span: stmt.span(),
+                            stmts: replacement,
+                        });
+                    }),
+                ]
+                .into()
+            }
+            AstPathRange::StartAfter(path) => {
+                let mut parent = &path[..];
+                while !parent.is_empty()
+                    && !matches!(parent.last().unwrap(), AstParentKind::Stmt(_))
+                {
+                    parent = &parent[0..parent.len() - 1];
+                }
+                if !parent.is_empty() {
+                    parent = &parent[0..parent.len() - 1];
+                    fn replace(stmts: &mut Vec<Stmt>, start_index: usize) {
+                        if stmts.len() > start_index + 1 {
+                            let unreachable = stmts
+                                .splice(
+                                    start_index + 1..,
+                                    [quote!("\"TURBOPACK unreachable\";" as Stmt)].into_iter(),
+                                )
+                                .collect::<Vec<_>>();
+                            for mut stmt in unreachable {
+                                ExtractDeclarations { stmts }.visit_mut_stmt(&mut stmt);
```

Comment:
Is this intentional? I'm asking this because I typically use `stmt.visit_mut_with(&mut ExtractDeclarations { stmts })` instead

Reply:
done